### ISSUE TITLE
ValueError for delete after x days

### ISSUE THREAD

**Issue by bbilly1** [bbilly1/tubearchivist#113]
Thank you so much for implementing this feature!
I am still having some trouble getting it to work properly though.

At first I was getting FileNotFound Exceptions because I manually deleted videos before and didnt rescan the filesystem after.
That I got sorted out.
The first run after that successfully deleted 20 videos, but now whenever this runs I get this error:

```
tubearchivist          | [2021-12-19 11:08:25,713: WARNING/ForkPoolWorker-4] get video data for rCdlBYUdTWQ
tubearchivist          | [2021-12-19 11:08:46,187: WARNING/ForkPoolWorker-4] auto delete older than True days
tubearchivist          | [2021-12-19 11:08:46,430: WARNING/ForkPoolWorker-4] add deleted to ignore list
tubearchivist          | [2021-12-19 11:08:46,526: WARNING/ForkPoolWorker-4] <Response [400]>
tubearchivist          | [2021-12-19 11:08:46,577: ERROR/ForkPoolWorker-4] Task download_pending[a9374e3f-1d52-4109-8de1-11d84f5d12e6] raised unexpected: ValueError('failed to add video to download queue')
tubearchivist          | Traceback (most recent call last):
tubearchivist          |   File "/usr/local/lib/python3.9/site-packages/celery/app/trace.py", line 451, in trace_task
tubearchivist          |     R = retval = fun(*args, **kwargs)
tubearchivist          |   File "/usr/local/lib/python3.9/site-packages/celery/app/trace.py", line 734, in __protected_call__
tubearchivist          |     return self.run(*args, **kwargs)
tubearchivist          |   File "/app/home/tasks.py", line 86, in download_pending
tubearchivist          |     downloader.run_queue()
tubearchivist          |   File "/app/home/src/download.py", line 565, in run_queue
tubearchivist          |     self.auto_delete_watched(autodelete_days)
tubearchivist          |   File "/app/home/src/download.py", line 804, in auto_delete_watched
tubearchivist          |     pending_handler.add_to_pending(all_youtube_ids, ignore=True)
tubearchivist          |   File "/app/home/src/download.py", line 98, in add_to_pending
tubearchivist          |     raise ValueError("failed to add video to download queue")
tubearchivist          | ValueError: failed to add video to download queue
```

_Originally posted by @cpt-kuesel in https://github.com/bbilly1/tubearchivist/issues/56#issuecomment-997367138_

**Comment by cpt-kuesel** [comment]
In the meantime I wanted to test my hunch that it might be videos downloaded before I updated to 0.9 that caused the above errors.
Started a new stack of TubeArchivist from scratch on my Synology instead of the server, downloaded my abos but the same ValueError occurs when the auto_delete gets triggered and no videos are getting deleted.

**Comment by bbilly1** [comment]
Sorry, been a bit slow on this one. 

Yeah, that was a classical example of quickly trying to fix something but only making it worse. It's connected to #112 and introduced there with the hotfix. 

I've also observed the same when I took another look at it and it's fixed here already: https://github.com/bbilly1/tubearchivist/commit/88ebb6d648cddef339613e37e535ffe9a96cd6f2 

But your observation is spot on: Only occurs when there are no videos to delete but ff there are videos within the date range it will delete them successfully. 

As far as I can tell the download process still downloads and indexes all pending videos without error even if it fails there when there is nothing to delete and that only occurs if you have the option enabled in the first place. 

As you already have a testing setup on your synology you can try to pull the `unstable` tag at let me know if that works as expected?

**Comment by cpt-kuesel** [comment]
Yes from what I can see, downloading and indexing is done before trying to auto download. And starting anew I did download 315 videos, which is 63 abos times page size 5. The error appeared afterwards.
Oh I will turn the setup on my server into the testing setup, the move to synology was planned once auto download + delete were implemented.
which by the way, the auto download is working very well, a really great addition!
